plugin-cgroupv2/src/k8s/pod.rs

```
@@ -0,0 +1,929 @@
+pub async fn get_pod_infos(uid: &str, node: &str, kubernetes_api_url: &str, token: &Token) -> anyhow::Result<PodInfos> {
+    let new_uid = uid.replace('_', "-");
+    let token = match token.get_value().await {
+        Ok(token) => token,
+        Err(e) => {
+            log::error!("Could not retrieve the token, got: {e}");
+            return Ok(PodInfos::empty());
+        }
+    };
+
+    if kubernetes_api_url.is_empty() {
+        return Ok(PodInfos::empty());
+    }
+
+    let mut api_url_root = kubernetes_api_url.to_string();
+    api_url_root.push_str("/api/v1/pods/");
+    let mut selector = false;
+
+    let api_url = if node.is_empty() {
+        api_url_root.to_owned()
+    } else {
+        let tmp = format!("{}?fieldSelector=spec.nodeName={}", api_url_root, node);
+        selector = true;
+        tmp
+    };
+
+    let mut headers = header::HeaderMap::new();
+    headers.insert(header::AUTHORIZATION, format!("Bearer {}", token).parse()?);
+    let client = reqwest::Client::builder()
+        .danger_accept_invalid_certs(true)
+        .default_headers(headers)
+        .build()?;
+
+    let Ok(response) = client.get(api_url).send().await else {
+        return Ok(PodInfos::empty());
+    };
+
+    let mut data: Value = match response.json().await {
+        Ok(value) => value,
+        Err(err) => {
+            log::error!("Error parsing JSON: {}", err);
+            Value::Null
+        }
+    };
+
+    // let's check if the items' part contain pods to look at
+    if let Some(items) = data.get("items") {
+        // If the node was not found i.e. no item in the response, we call the API again with all nodes
+        let size = items.as_array().map(|a| a.len()).unwrap_or(0);
+        if size == 0 && selector {
+            // Ask again the api, with all nodes
+            let Ok(response) = client.get(api_url_root).send().await else {
+                return Ok(PodInfos::empty());
+            };
```

Does it swallow the error?